airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/publish/context.py

```diff
@@ -50,6 +51,9 @@ def __init__(
         pipeline_name = f"Publish {connector.technical_name}"
         pipeline_name = pipeline_name + " (pre-release)" if pre_release else pipeline_name
 
+        if use_local_cdk and not self.pre_release:
+            raise ValueError("Publishing with the local CDK is only supported for pre-release publishing.")
```

Comment:
Not sure if there is a better place to perform this validation? In the publish pipeline perhaps?